modules/javafx.graphics/src/main/java/com/sun/javafx/application/LauncherImpl.java

```diff
@@ -575,38 +574,31 @@ private static ClassLoader setupJavaFXClassLoader(File appJar, String fxClassPat
         return null;
     }
 
-    private static String decodeBase64(String inp) throws IOException {
+    private static String decodeBase64(String inp) {
         return new String(Base64.getDecoder().decode(inp));
     }
 
     private static String[] getAppArguments(Attributes attrs) {
         List args = new LinkedList();
 
-        try {
-            int idx = 1;
-            String argNamePrefix = MF_JAVAFX_ARGUMENT_PREFIX;
-            while (attrs.getValue(argNamePrefix + idx) != null) {
-                args.add(decodeBase64(attrs.getValue(argNamePrefix + idx)));
-                idx++;
-            }
-
-            String paramNamePrefix = MF_JAVAFX_PARAMETER_NAME_PREFIX;
-            String paramValuePrefix = MF_JAVAFX_PARAMETER_VALUE_PREFIX;
-            idx = 1;
-            while (attrs.getValue(paramNamePrefix + idx) != null) {
-                String k = decodeBase64(attrs.getValue(paramNamePrefix + idx));
-                String v = null;
-                if (attrs.getValue(paramValuePrefix + idx) != null) {
-                    v = decodeBase64(attrs.getValue(paramValuePrefix + idx));
-                }
-                args.add("--" + k + "=" + (v != null ? v : ""));
-                idx++;
-            }
-        } catch (IOException ioe) {
-            if (verbose) {
-                System.err.println("Failed to extract application parameters");
-            }
-            ioe.printStackTrace();
+        int idx = 1;
+        String argNamePrefix = MF_JAVAFX_ARGUMENT_PREFIX;
+        while (attrs.getValue(argNamePrefix + idx) != null) {
+            args.add(decodeBase64(attrs.getValue(argNamePrefix + idx)));
+            idx++;
+        }
+
+        String paramNamePrefix = MF_JAVAFX_PARAMETER_NAME_PREFIX;
+        String paramValuePrefix = MF_JAVAFX_PARAMETER_VALUE_PREFIX;
+        idx = 1;
+        while (attrs.getValue(paramNamePrefix + idx) != null) {
+            String k = decodeBase64(attrs.getValue(paramNamePrefix + idx));
+            String v = null;
+            if (attrs.getValue(paramValuePrefix + idx) != null) {
+                v = decodeBase64(attrs.getValue(paramValuePrefix + idx));
+            }
+            args.add("--" + k + "=" + (v != null ? v : ""));
+            idx++;
```

Comment:
The code was already broken (since 2014 apparently when the new Base64 decoder was used).  Had the author turned on these handy warnings, they would have noticed that something was amiss.  Now they unwittingly changed the behavior.

Personally, I think the current code is an improvement.  The catch block was already unreachable since 2014, and the `IllegalArgumentException` has been bubbling up and terminating the program since then (it's called from `launchApplication`).

Whether a follow-up bug is needed depends on whether we want to restore the original behavior: log an exception, ignore parsing the rest of the parameters and continue without them...  I think we shouldn't.

Reply:
I agree with you - we shouldn't.

Reply:
This is an example of what I was talking about in [this comment](https://github.com/openjdk/jfx/pull/958#pullrequestreview-1198517697).

Reply:
I agree that no follow-on bug is needed.